src/painter.rs

```diff
@@ -35,6 +35,54 @@ impl PromptCoordinates {
     }
 }
 
+pub struct PromptLines<'prompt> {
```

Comment:
Bikeshed: `BufferView` as it does not care about the `Prompt` directly and `View` may reflect the reference nature. Same for the lifetime name